src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/LeftCurlyCheck.java

```diff
@@ -277,7 +277,7 @@ private static DetailAST skipAnnotationOnlyLines(DetailAST ast) {
                     tokenAfterLast = lastAnnotation.getNextSibling();
                 }
 
-                if (tokenAfterLast.getLineNo() > lastAnnotation.getLineNo()) {
+                if (tokenAfterLast.getLineNo() != lastAnnotation.getLineNo()) {
```

Comment:
This area is weird.
`>=` showed no differences in regression.
If I removed entire if/else and replaced it with `resultNode = tokenAfterLast;`, regression showed no differences again.